src/drive/lib/encryption.js

```diff
@@ -1,3 +1,80 @@
+const stringToArrayBuffer = string => {
+  var encoder = new TextEncoder('utf-8')
+  return encoder.encode(string)
+}
+
+const asDerivableKey = async (
+  baseMaterial,
+  keyDerivationAlgorithm = 'PBKDF2'
+) => {
+  let usableMaterial
+  if (baseMaterial instanceof CryptoKey) {
+    usableMaterial = await window.crypto.subtle.exportKey('raw', baseMaterial)
+  } else {
+    usableMaterial = stringToArrayBuffer(baseMaterial)
+  }
+  return await window.crypto.subtle.importKey(
+    'raw',
+    usableMaterial,
+    { name: keyDerivationAlgorithm },
+    true,
+    ['deriveKey']
+  )
+}
+
+const slowHashing = async (
+  baseKey,
+  saltBuffer,
+  { keyDerivationAlgorithm, iterations, hash } = {},
+  { algorithm, keyLength } = {}
+) => {
+  return await window.crypto.subtle.deriveKey(
+    {
+      name: keyDerivationAlgorithm || 'PBKDF2',
+      salt: saltBuffer,
+      iterations: iterations || 1000,
```

Comment:
Why 1000? We specified 100k iterations in our paper, is it too time-consuming?

Reply:
No, my mistake. I changed it but forgot this one